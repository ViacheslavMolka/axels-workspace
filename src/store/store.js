import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from 'redux';

import reducer, { watchLoadData } from '../redux/ducks/orders';


const saga = createSagaMiddleware();

export const store = createStore(reducer, applyMiddleware(saga));

saga.run(watchLoadData);
