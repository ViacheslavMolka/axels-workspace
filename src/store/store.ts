import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';

import reducer, { watchLoadData } from '../redux/ducks/orders';


const saga = createSagaMiddleware();

export const store = configureStore({
    reducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(saga)
}) 

export type RootState = ReturnType<typeof store.getState>;

saga.run(watchLoadData);
