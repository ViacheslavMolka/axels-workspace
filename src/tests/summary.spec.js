import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../store/store';

import SummaryCompoment from '../components/shoppingCard';


describe('SummaryCompoment component', () => {
    it('SummaryCompoment snapshot', () => {
        const component = renderer.create(<Provider store={store}><SummaryCompoment/></Provider>).toJSON()
        expect(component).toMatchSnapshot()
    });
});
