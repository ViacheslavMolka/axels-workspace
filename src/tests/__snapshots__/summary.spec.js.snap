// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCompoment component SummaryCompoment snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-4"
  >
    <ol
      className="sc-dlfnbm ffohbB"
    >
      <li
        className="breadcrumb-item"
        onClick={[Function]}
      >
        <a
          href="/shipping/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Shipping
        </a>
      </li>
      <li
        className="breadcrumb-item"
        onClick={[Function]}
      >
        <a
          href="/shipping/billing/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Billing
        </a>
      </li>
      <li
        className="breadcrumb-item"
        onClick={[Function]}
      >
        <a
          href="/shipping/billing/payment/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Payment
        </a>
      </li>
    </ol>
    <div
      className="sc-jrAGrp hjHatU"
    />
  </div>
  <div
    className="col-sm-3"
  >
    <div
      className=""
    />
    <div>
      <div
        className="sc-kEjbxe fA-dwZJ"
      >
        <label>
          Order Summary
        </label>
        <a
          href="/#"
        >
          edit order
        </a>
      </div>
      <div
        className="sc-crrsfI hQcpOD"
      >
        <div
          className="list-group"
          onKeyDown={[Function]}
        >
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Subtotal
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Shipping
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Taxes
          </div>
        </div>
        <div
          className="list-group"
          onKeyDown={[Function]}
        >
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            $
            0
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Free
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            $10
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-dQppl bxdFGW bottom-wrapp"
    >
      <div
        className="total-price"
      >
        <label>
          Total
        </label>
        <label>
          $
          10
        </label>
      </div>
      <div
        className="footer-text"
      >
        All purchases are subject to our Terms and Conditions
      </div>
    </div>
  </div>
</div>
`;

exports[`SummaryCompoment snapshot 1`] = `ShallowWrapper {}`;
